pkg/cloud_provider/clientset/clientset.go

```diff
@@ -78,13 +78,47 @@ func New(kubeconfigPath string, informerResyncDurationSec int) (Interface, error
 }
 
 func (c *Clientset) ConfigurePodLister(nodeName string) {
-	trimManagedFields := func(obj interface{}) (interface{}, error) {
+	trim := func(obj interface{}) (interface{}, error) {
 		if accessor, err := meta.Accessor(obj); err == nil {
 			if accessor.GetManagedFields() != nil {
 				accessor.SetManagedFields(nil)
 			}
 		}
 
+		podObj, ok := obj.(*corev1.Pod)
+		if !ok {
+			return obj, nil
+		}
+
```

Comment:
put comments around why we are only filtering relevant parts of the pod spec, link to the node.go csi node publish